src/api/route-services/samples.js

```diff
@@ -9,37 +11,39 @@ class SamplesService {
     this.tableName = `samples-${config.clusterEnv}`;
   }
 
-  async getSamples(experimentId) {
-    const key = convertToDynamoDbRecord({
-      experimentId,
+  async getSamples(projectUuid) {
+    const marshalledData = convertToDynamoDbRecord({
+      ':projectUuid': projectUuid,
     });
 
     const params = {
       TableName: this.tableName,
-      Key: key,
+      IndexName: 'gsiExperimentid',
+      KeyConditionExpression: 'projectUuid = :projectUuid',
+      ExpressionAttributeValues: marshalledData,
       ProjectionExpression: 'samples',
     };
     const dynamodb = createDynamoDbInstance();
 
-    const response = await dynamodb.getItem(params).promise();
+    const response = await dynamodb.query(params).promise();
 
     if (response.Item) {
       const prettyResponse = convertToJsObject(response.Item);
       return prettyResponse;
     }
 
-    throw Error('Sample not found');
+    throw new NotFoundError('Samples not found');
   }
 
-  async getSampleIds(experimentId) {
-    const key = convertToDynamoDbRecord({
+  async getByExperimentId(experimentId) {
```

Comment:
if you are returning samples for an experiment, it is worth renaming that function to something like `getSampleIdsForExperiment` to make it more explicit